Makefile

```diff
@@ -42,71 +63,90 @@ lint: $(TOOLS_DIR)/golangci-lint $(TOOLS_DIR)/misspell
 	$(TOOLS_DIR)/golangci-lint run --timeout=5m -v
 	$(TOOLS_DIR)/misspell $(ALL_DOCS)
 
+.PHONY: unit-test
+unit-test:
+	go test $(GO_FILES) -v
+
 .PHONY: build
-build: setup
-	@echo "Building..."
-	$Q GOOS=linux CGO_ENABLED=0 go build -ldflags ${LDFLAGS} -o _output/secrets-store-csi-driver-provider-azure ./cmd/
+build:
+	CGO_ENABLED=0 GOOS=linux go build -a -ldflags ${LDFLAGS} -o _output/secrets-store-csi-driver-provider-azure ./cmd/
 
 .PHONY: build-windows
 build-windows:
-	@echo "Building windows binary..."
 	CGO_ENABLED=0 GOOS=windows go build -a -ldflags ${LDFLAGS} -o _output/secrets-store-csi-driver-provider-azure.exe ./cmd/
 
-image:
-	@echo "Building docker image..."
-	docker buildx build --no-cache -t $(DOCKER_IMAGE):$(IMAGE_VERSION) -f Dockerfile --platform="linux/amd64" --output "type=docker,push=false" .
-
-.PHONY: build-container-windows
-build-container-windows:
-	@echo "Building windows docker image..."
-	docker buildx build --no-cache -t $(DOCKER_IMAGE):$(IMAGE_VERSION) -f windows.Dockerfile --platform="windows/amd64" --output "type=docker,push=false" .
-
-push: image
-	docker push $(DOCKER_IMAGE):$(IMAGE_VERSION)
-
-setup: clean
-	@echo "Setup..."
-	$Q go env
-
+.PHONY: build-darwin
+build-darwin:
+	CGO_ENABLED=0 GOOS=darwin go build -a -ldflags ${LDFLAGS} -o _output/secrets-store-csi-driver-provider-azure.exe ./cmd/
+
+.PHONY: container
+container: build
+	docker build --no-cache -t $(IMAGE_TAG) -f Dockerfile .
+
+.PHONY: container-linux
+container-linux: docker-buildx-builder
+	docker buildx build --no-cache --output=type=$(OUTPUT_TYPE) --platform="linux/$(ARCH)" \
+			-t $(IMAGE_TAG)-linux-$(ARCH) -f Dockerfile .
+
+.PHONY: container-windows
+container-windows: docker-buildx-builder
+	docker buildx build --no-cache --output=type=$(OUTPUT_TYPE) --platform="windows/amd64" \
+	 		-t $(IMAGE_TAG)-windows-$(OSVERSION)-$(ARCH) --build-arg OSVERSION=$(OSVERSION) -f windows.Dockerfile .
+
+.PHONY: docker-buildx-builder
+docker-buildx-builder:
+	@if ! DOCKER_CLI_EXPERIMENTAL=enabled docker buildx ls | grep -q container-builder; then\
+		DOCKER_CLI_EXPERIMENTAL=enabled docker buildx create --name container-builder --use --driver-opt image=moby/buildkit:v0.7.2;\
```

Comment:
lets make buildkit version a variable

Reply:
Done! Also updated to `buildkit:v0.8.1` which contains the fix for windows image builds.